src/sentry/integrations/slack/views/link_team.py

```diff
@@ -140,79 +130,78 @@ def handle(self, request: Request, signed_params: str) -> Response:
                 },
             )
 
-        team = Team.objects.get(id=team_id, organization=organization)
-        if not team:
-            return self.render_error_page(body_text="HTTP 404: Team does not exist")
-
-        INSUFFICIENT_ROLE_MESSAGE = {
-            "heading": "Insufficient role",
-            "body": "You must be an admin or higher to link teams.",
-        }
-        ALREADY_LINKED_MESSAGE = {
-            "heading": "Already linked",
-            "body": f"The {team.slug} team has already been linked to a Slack channel.",
-        }
-        SUCCESS_LINKED_MESSAGE = {
-            "heading": "Team linked",
-            "body": f"The {team.slug} team will now receive issue alert notifications in the {channel_name} channel.",
-        }
-        identity = self.get_identity(request, integration, params["slack_id"])
+        if not form.is_valid():
```

Comment:
I'm concerned about removing the check for it being a POST request because validity is determined in `/sentry/.venv/lib/python3.6/site-packages/django/forms/forms.py`. On the GET request the form is always invalid because there is no data so `self.is_bound` is False. A form is valid if it has data and that data has no errors in it.

Reply:
If the method is `GET` then we will always have responded on line 123.

Reply:
ah oops! I missed that, the diff is a little hard to read